src/app/services/code_service.js

```diff
@@ -56,6 +56,28 @@ angular
         return query.join("\n");
     }
 
+    service.generateMetricSQL = function(metric) {
+        if (metric.type == 'expression') {
+            return metric.sql
+        } else {
+            let query_parts = [
+                `select ${metric.type}(${metric.sql})` ,
+                `from {{ ${metric.model} }}`,
+            ]
+
+            if (metric.filters.length > 0) {
+                let filter_exprs = _.map(metric.filters, (filter) => {
+                    return `${filter.field} ${filter.operator} ${filter.value}`
+                })
+
+                let filters = filter_exprs.join(" AND ")
+                query_parts.push(`where ${filters}`)
+            }
+
+            return query_parts.join("\n")
+        }
+    }
```

Comment:
Unnecessary else, using mixed single and double quotes, using mixed snake and camel case, unnecessary use of lodash, let instead of const
```suggestion
    service.generateMetricSQL = function(metric) {
        if (metric.type == 'expression') {
            return metric.sql
        }

        const queryParts = [
            `select ${metric.type}(${metric.sql})` ,
            `from {{ ${metric.model} }}`,
        ];

        if (metric.filters.length > 0) {
            const filterExprs = metric.filters.map(filter => (
                `${filter.field} ${filter.operator} ${filter.value}`
            ))

            const filters = filterExprs.join(' AND ')
            queryParts.push(`where ${filters}`)
        }

        return queryParts.join('\n')
    }
```